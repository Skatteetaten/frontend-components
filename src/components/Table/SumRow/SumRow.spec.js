import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SumRow } from '.';

describe('SumRow komponent', () => {
  describe('Når SumRow består av 0 kolonner', () => {
    const wrapper = mount(
      <SumRow
        numberOfColumns={0}
        sum={{ text: 'sumtext', colspan: 2, total: 153 }}
      />
    );

    it('Så ser markupen riktig ut', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      // Det skal bare vises 1 for sumteksten og 1 td for selve summen
      expect(wrapper.find('th').length).toEqual(1);
      expect(wrapper.find('td').length).toEqual(1);
    });
  });

  describe('Når SumRow består av f.eks 5 kolonner', () => {
    const wrapper = mount(
      <SumRow
        numberOfColumns={5}
        sum={{ text: 'sumtext', colspan: 2, total: 153 }}
      />
    );

    it('Så ser markupen riktig ut', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      // Det skal bare vises 1 for sumteksten og 3 td for å fylle ut bredden
      expect(wrapper.find('th').length).toEqual(1);
      expect(wrapper.find('td').length).toEqual(3);
    });
  });

  describe('Når SumRow har en editableRows i tillegg', () => {
    const wrapper = mount(
      <SumRow
        numberOfColumns={0}
        editableRows
        sum={{ text: 'sumtext', colspan: 2, total: 153 }}
      />
    );
    it('Så ser markupen riktig ut', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      // Det skal bare vises 1 for sumteksten og 2 td: én for selve summen og én for editableRows
      expect(wrapper.find('th').length).toEqual(1);
      expect(wrapper.find('td').length).toEqual(2);
    });
  });

  describe('Når SumRow har en expandableRows i tillegg', () => {
    it('Så ser markupen riktig ut uten expandableRows', () => {
      const wrapper = mount(
        <SumRow
          numberOfColumns={0}
          expandableRows
          sum={{ text: 'sumtext', colspan: 2, total: 153 }}
        />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
      // Det skal bare vises 1 for sumteksten og 1 td for summen
      expect(wrapper.find('th').length).toEqual(1);
      expect(wrapper.find('td').length).toEqual(1);
    });

    it('Så ser markupen riktig ut med expandIconPlacement === before', () => {
      const wrapper = mount(
        <SumRow
          numberOfColumns={0}
          expandableRows
          expandIconPlacement={'before'}
          sum={{ text: 'sumtext', colspan: 2, total: 153 }}
        />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
      // Det skal bare vises 1 for sumteksten og 1 td for summen
      expect(wrapper.find('th').length).toEqual(1);
      expect(wrapper.find('td').length).toEqual(1);
    });

    it('Så ser markupen riktig ut med expandIconPlacement === after', () => {
      const wrapper = mount(
        <SumRow
          numberOfColumns={0}
          expandableRows
          expandIconPlacement={'after'}
          sum={{ text: 'sumtext', colspan: 2, total: 153 }}
        />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
      // Det skal bare vises 1 for sumteksten og 2 td: én for summen og én for expandableRows
      expect(wrapper.find('th').length).toEqual(1);
      expect(wrapper.find('td').length).toEqual(2);
    });
  });
});
