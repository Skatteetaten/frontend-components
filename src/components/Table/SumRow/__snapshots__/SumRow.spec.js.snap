// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SumRow komponent Når SumRow består av 0 kolonner Så ser markupen riktig ut 1`] = `
<SumRow
  numberOfColumns={0}
  sum={
    Object {
      "colspan": 2,
      "text": "sumtext",
      "total": 153,
    }
  }
>
  <tr
    className=

        {
          vertical-align: middle;
        }
  >
    <th
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            font-weight: 700;
            text-align: end;
          }
      colSpan={2}
      scope="row"
    >
      sumtext
    </th>
    <td
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            text-align: end;
          }
    >
      153
    </td>
  </tr>
</SumRow>
`;

exports[`SumRow komponent Når SumRow består av f.eks 5 kolonner Så ser markupen riktig ut 1`] = `
<SumRow
  numberOfColumns={5}
  sum={
    Object {
      "colspan": 2,
      "text": "sumtext",
      "total": 153,
    }
  }
>
  <tr
    className=

        {
          vertical-align: middle;
        }
  >
    <th
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            font-weight: 700;
            text-align: end;
          }
      colSpan={2}
      scope="row"
    >
      sumtext
    </th>
    <td
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            text-align: end;
          }
    >
      153
    </td>
    <td
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
      key="0"
    />
    <td
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
      key="1"
    />
  </tr>
</SumRow>
`;

exports[`SumRow komponent Når SumRow har en editableRows i tillegg Så ser markupen riktig ut 1`] = `
<SumRow
  editableRows={true}
  numberOfColumns={0}
  sum={
    Object {
      "colspan": 2,
      "text": "sumtext",
      "total": 153,
    }
  }
>
  <tr
    className=

        {
          vertical-align: middle;
        }
  >
    <th
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            font-weight: 700;
            text-align: end;
          }
      colSpan={2}
      scope="row"
    >
      sumtext
    </th>
    <td
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            text-align: end;
          }
    >
      153
    </td>
    <td />
  </tr>
</SumRow>
`;

exports[`SumRow komponent Når SumRow har en expandableRows i tillegg Så ser markupen riktig ut med expandIconPlacement === after 1`] = `
<SumRow
  expandIconPlacement="after"
  expandableRows={true}
  numberOfColumns={0}
  sum={
    Object {
      "colspan": 2,
      "text": "sumtext",
      "total": 153,
    }
  }
>
  <tr
    className=

        {
          vertical-align: middle;
        }
  >
    <th
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            font-weight: 700;
            text-align: end;
          }
      colSpan={2}
      scope="row"
    >
      sumtext
    </th>
    <td
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            text-align: end;
          }
    >
      153
    </td>
    <td />
  </tr>
</SumRow>
`;

exports[`SumRow komponent Når SumRow har en expandableRows i tillegg Så ser markupen riktig ut med expandIconPlacement === before 1`] = `
<SumRow
  expandIconPlacement="before"
  expandableRows={true}
  numberOfColumns={0}
  sum={
    Object {
      "colspan": 2,
      "text": "sumtext",
      "total": 153,
    }
  }
>
  <tr
    className=

        {
          vertical-align: middle;
        }
  >
    <th
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            font-weight: 700;
            text-align: end;
          }
      colSpan={2}
      scope="row"
    >
      sumtext
    </th>
    <td
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            text-align: end;
          }
    >
      153
    </td>
  </tr>
</SumRow>
`;

exports[`SumRow komponent Når SumRow har en expandableRows i tillegg Så ser markupen riktig ut uten expandableRows 1`] = `
<SumRow
  expandableRows={true}
  numberOfColumns={0}
  sum={
    Object {
      "colspan": 2,
      "text": "sumtext",
      "total": 153,
    }
  }
>
  <tr
    className=

        {
          vertical-align: middle;
        }
  >
    <th
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            font-weight: 700;
            text-align: end;
          }
      colSpan={2}
      scope="row"
    >
      sumtext
    </th>
    <td
      className=

          {
            font-weight: 400;
            padding-bottom: 1rem;
            padding-left: 1rem;
            padding-right: 1rem;
            padding-top: 1rem;
            vertical-align: inherit;
          }
          {
            text-align: end;
          }
    >
      153
    </td>
  </tr>
</SumRow>
`;
